### ISSUE TITLE
GTA: San Andreas - The Definitive Edition crashes with HDR on.

### ISSUE THREAD

**Issue by Hadrianneue** [doitsujin/dxvk#4449]
Game crashes with just the DXVK_HDR=1 with fatal error dialog message on startup, crashes with gamescope if using the --hdr-enabled option, by default it runs with DX11, forcing DX12 with %command% -dx12 actually gets in game without crashes, allows me to enabled HDR but it doesn't work, has no effect.

originally submitted on https://github.com/ValveSoftware/gamescope/issues/1625

### Software information
- Desktop environment: KDE 6.2
- Session type: wayland
- Gamescope version: 3.15.9-21-g7dd1bcd9 (gcc 14.2.1)

### System information
- GPU: RX 6700 XT
- Driver: mesa-git
- Wine version: proton experimental bleeding-edge
- DXVK version: dxvk-git

### Log files
[steam.log](https://github.com/user-attachments/files/17722567/steam.log)

**Comment by doitsujin** [comment]
This is unfortunately normal for UE4 since UE4 only supports HDR through AGS or NVAPI and doesn't error-check a single thing, so if anything goes wrong there it just explodes. We have code to disable HDR for UE4 but it doesn't work here because the exe name doesn't seem to follow the usual `*-Shipping.exe` pattern.

**Comment by Blisto91** [comment]
You can try to launch with `PROTON_FORCE_NVAPI=1 %command% to see if that allows it to work in d3d11 mode. It not working in d3d12 mode i don't know but it could just be a game thing as dx12 might not be considered supported or tested by the devs.

**Comment by Hadrianneue** [comment]
> You can try to launch with `PROTON_FORCE_NVAPI=1 %command%` to see if that allows it to work in d3d11 mode. It not working in d3d12 mode i don't know but it could just be a game thing as dx12 might not be considered supported or tested by the devs.
> 
> Edit: To clarify. The reason the ags path doesn't work with these UE4 games is that it is statically linked which doesn't work in Proton/Wine.

that indeed works and is necessary on asseto corsa competizione, which is dx11

**Comment by Hadrianneue** [comment]
> This is unfortunately normal for UE4 since their DX11 path only supports HDR through AGS or NVAPI for some reason and doesn't error-check a single thing, so if anything goes wrong there it just explodes. We have code to disable HDR for UE4 but it doesn't work here because the exe name doesn't seem to follow the usual `*-Shipping.exe` pattern.
> 
> TL;DR if you want to use HDR with any Unreal Engine game, you **have** to use D3D12, but we can add an app profile to avoid the crash.

PROTON_FORCE_NVAPI=1 does prevent crashing even on dx11, either way HDR is simply not working, the toggle makes no difference... other UE4/5 games work just fine with HDR

Actually im going to check windows later today, not even sure if its working there

**Comment by qinlili23333** [comment]
> because the exe name doesn't seem to follow the usual `*-Shipping.exe` pattern

Maybe we should improve UE4 detecting method. 
Current status is:
- Some game doesn't follow Shipping.exe format (GTASA-DE in this issue, and Hi-Fi Rush on my device is `Hi-Fi-RUSH.exe`)
- Some UE3 game also follow Shipping.exe format (`MMH7Game-Win64-Shipping.exe` for Might and Magic Heroes VII)
- We hardcoded `bool isUE4 = exeName.find("-Win64-Shipping") != std::string::npos;` which won't allow 32-bit games been detected, but there are some 32-bit UE4 games (I made[ a list of them](https://www.pcgamingwiki.com/wiki/User:Qinlili/List_of_Windows_32-bit_UE4_games) using PCGW query, although I didn't see any of them has HDR support currently)

I suggest to use `\Content\Paks` folder structure to detect UE4.

**Comment by Blisto91** [comment]
@Hadrianneue Did you find out if HDR works on Windows?

**Comment by mirh** [comment]
> since their DX11 path only supports HDR through AGS or NVAPI for some reason

Shouldn't this be handled by/in dxvk-nvapi then?

**Comment by Blisto91** [comment]
No as a user might not be using nvapi.

**Comment by mirh** [comment]
Dxvk isn't supposed to handle d3d12.
